### ISSUE TITLE
cart2pol returns sqrt(x^2 + y^2 + eps): rho(0,0) = 1e-4, rho(3,4) = 5.000000001, and the eps docstring describes a division that does not exist

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3939]
`cart2pol` folds its `eps` into the radius **value** rather than only into the gradient path (`kornia/geometry/conversions.py:167-169`):

```python
rho = torch.sqrt(x**2 + y**2 + eps)
phi = torch.atan2(y, x)
```

Two problems follow. First, the returned `rho` is never the true radius: at the origin it is `sqrt(1e-8) = 1e-4` instead of `0`, and everywhere else it carries a systematic `+eps/(2ρ)` bias — `cart2pol(3., 4.)` returns `5.000000001` in float64, not `5`. The absolute error at the origin (`1e-4`) is six orders of magnitude larger than `eps` itself, because `sqrt` amplifies it. Second, the docstring justifies the parameter as `eps: To avoid division by zero.`, but the function body contains **no division at all** — the epsilon's real and only effect is to keep `d(rho)/d(x, y)` finite at the origin.

## Repro

```python
import torch
from kornia.geometry.conversions import cart2pol

f64 = torch.float64
print("cart2pol(0., 0.)          ->", [t.item() for t in cart2pol(torch.tensor(0.), torch.tensor(0.))])
print("cart2pol(0., 0.) f64 rho  ->", repr(cart2pol(torch.tensor(0., dtype=f64), torch.tensor(0., dtype=f64))[0].item()))
r, _ = cart2pol(torch.tensor(3., dtype=f64), torch.tensor(4., dtype=f64))
print("cart2pol(3., 4.) f64 rho  ->", repr(r.item()), "| err vs 5.0:", r.item() - 5.0)
print("torch.hypot(3., 4.) f64   ->", torch.hypot(torch.tensor(3., dtype=f64), torch.tensor(4., dtype=f64)).item())
print("cart2pol(0., 0., eps=0.0) ->", [t.item() for t in cart2pol(torch.tensor(0.), torch.tensor(0.), eps=0.0)])

for eps in (1e-8, 0.0):
    xy = torch.zeros(2, dtype=f64, requires_grad=True)
    rho, _ = cart2pol(xy[0], xy[1], eps=eps)
    rho.backward()
    print(f"grad of rho at the origin, eps={eps}: {xy.grad.tolist()}")
```

## Observed

```
cart2pol(0., 0.)          -> [9.999999747378752e-05, 0.0]
cart2pol(0., 0.) f64 rho  -> 0.0001
cart2pol(3., 4.) f64 rho  -> 5.000000001 | err vs 5.0: 1.000000082740371e-09
torch.hypot(3., 4.) f64   -> 5.0
cart2pol(0., 0., eps=0.0) -> [0.0, 0.0]
grad of rho at the origin, eps=1e-08: [0.0, 0.0]
grad of rho at the origin, eps=0.0: [nan, nan]
```

## Expected

- `cart2pol(0., 0.)[0]` should be `0.0`. The polar radius of the origin is zero; `1e-4` is not a rounding artifact, it is a value a user can trip over (e.g. any "is this point at the origin" test, or a subsequent `pol2cart` round-trip).
- `cart2pol(3., 4.)[0]` should be exactly `5.0` in float64, as `torch.hypot` gives. Observed `5.000000001`. The bias is not confined to degenerate inputs — it is present for every input, and its relative size grows as `rho` shrinks.
- The `eps` argument should protect the **gradient**, not perturb the **value**. The last two lines above show what it actually does: with `eps=0.0` the value is exact and the gradient at the origin is `[nan, nan]`; with the default `eps=1e-8` the gradient is finite (`[0.0, 0.0]`) and the value is wrong. Those two goals are separable. A double-`where` on the squared radius gives both:

```python
def rho(x, y, eps=1e-12):
    s = x**2 + y**2
    safe = torch.where(s > eps, s, torch.ones_like(s))
    return torch.where(s > eps, torch.sqrt(safe), torch.zeros_like(s))
```

Executed in float64: `rho(0., 0.) -> 0.0` with gradient `[0.0, 0.0]`, and `rho(3., 4.) -> 5.0` exactly. (Note that plain `torch.hypot` is *not* sufficient on its own — it returns the exact values `0.0` and `5.0`, but its gradient at the origin is still `[nan, nan]`, executed.)

Independently of the numeric fix, the `eps` docstring line (`To avoid division by zero.`) is wrong and should say what the parameter actually does.

## Classification

Mixed, and worth an explicit maintainer call:

- The **value bias** is a broken-output defect — `rho(3, 4) != 5` is wrong under any convention, and `rho(0, 0) = 1e-4` is a wrong answer rather than an alternative one.
- However, fixing it **changes the default output of every call**, including the `eps=1e-8` default path, so it cannot be landed with byte-identity discipline. In particular `rho(0, 0)` moves `1e-4 -> 0`, which is a visible, user-observable default change, and any caller currently relying on `rho > 0` at the origin would be affected.

Per the reference-core lifecycle (Part I) my reading is that the value fix belongs in the coordinated **Repair & Deprecation window** alongside the other semantic/default repairs, rather than landing early as an ordinary broken-output bugfix — but the argument for treating it as broken-output is strong enough that I am flagging the choice rather than deciding it. The docstring correction is unambiguous and is being handled separately in the conversions convention-block docs PR.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5).
